core/trino-main/src/main/java/io/trino/server/remotetask/TaskInfoFetcher.java

```diff
@@ -161,6 +167,17 @@ public void addFinalTaskInfoListener(StateChangeListener<TaskInfo> stateChangeLi
         fireOnceStateChangeListener.stateChanged(finalTaskInfo.get());
     }
 
+    public SpoolingOutputStats.Snapshot retrieveSpoolingOutputStats()
+    {
+        Optional<TaskInfo> finalTaskInfo = this.finalTaskInfo.get();
+        checkState(finalTaskInfo.isPresent(), "finalTaskInfo must be present");
+        TaskState taskState = finalTaskInfo.get().getTaskStatus().getState();
+        checkState(taskState == TaskState.FINISHED, "task must be FINISHED, got: %s", taskState);
+        SpoolingOutputStats.Snapshot result = spoolingOutputStats.getAndSet(null);
```

Comment:
Added a comment at the `RemoteTask#retrieveAndDropSpoolingOutputStats` java doc

Reply:
It is probalby obvious, but how are we sure that `TaskInfoFetcher.updateTaskInfo` will no longer be called and would not reset the `spoolingOutputStats`? Is that just because task is already done?

Reply:
I guess this is this code:
```
        if (updated && newValue.getTaskStatus().getState().isDone()) {
            finalTaskInfo.compareAndSet(Optional.empty(), Optional.of(newValue));
            stop();
        }
```